### PR TITLE
Backport fixes from two to three

### PR DESCRIPTION
**Pull request by albertvaka**
It would be much easier if we did these in parallel :/

Review commit by commit with the correspondent PR from `two.cpp` side-by-side for maximum happiness.

### CODE REVIEW

File: six/three/three.cpp [hunk 1]
```diff
@@ -117,11 +117,15 @@ bool Three::init()
...
+    if (init_stringutils() != EXIT_SUCCESS) {
+        goto done;
+    }
+
     // import the base class
     _baseClass = _importFrom("datadog_checks.checks", "AgentCheck");
 
 done:
-    // save thread state and release the GIL
+    // save tread state and release the GIL
```
Comment (KSerrania) [inline six/three/three.cpp]: Typo here 😄

Comment (albertvaka) [reply six/three/three.cpp]: Fixed!

File: six/three/three.cpp [hunk 2]
```diff
@@ -522,32 +596,52 @@ std::string Three::_fetchPythonError() const
...
-                        char *item = as_string(PyList_GetItem(fmt_exc, i));
+                    for (int i = 0; i < len; i++) {
+                        PyObject *s = PyList_GetItem(fmt_exc, i); // borrowed ref
+                        if (s == NULL || !PyString_Check(s)) {
+                            // unlikely to happen but same as above, do not propagate this error upstream
+                            // to avoid confusing the caller. PyErr_Clear() will be called before returning.
+                            ret_val = "";
+                            goto done;
+                        }
+                        char *item = as_string(s);
```
Comment (truthbk) [inline six/three/three.cpp]:
```suggestion
                        // traceback.format_exception returns a list of strings, each ending in a *newline* 
                        // and some containing internal newlines. No need to add any CRLF/newlines.
                        char *item = as_string(s);
```

**Review by truthbk — DISMISSED** [review]
Looks good to me, just added a small suggestion (code comment) on a minor thing that was unclear to me.

File: six/three/three.cpp (hunk 2 above)
Comment (albertvaka) [reply six/three/three.cpp]: I've added it two two.cpp as well.

**Review by truthbk — APPROVED** [review]
🙇

**Pull request merged**